src/generic_submitter.rs

```diff
@@ -25,6 +25,7 @@ pub fn share_submitted(_state: &GenericSubmitterState, user_id: &Vec<u8>, user_t
 	println!("Got valid share with value {} from \"{}\" from machine identified as \"{}\"", value, String::from_utf8_lossy(user_id), String::from_utf8_lossy(user_tag_1));
 }
 
-pub fn weak_block_submitted(_state: &GenericSubmitterState, user_id: &Vec<u8>, user_tag_1: &Vec<u8>, value: u64, _header: &BlockHeader, txn: &Vec<Vec<u8>>, _extra_block_data: &Vec<u8>, _leading_zeros: u8, _required_leading_zeros: u8) {
+pub fn weak_block_submitted(_state: &GenericSubmitterState, user_id: &Vec<u8>, user_tag_1: &Vec<u8>, value: u64, _header: &BlockHeader, txn: &Vec<Vec<u8>>, _extra_block_data: &Vec<u8>,
+	_leading_zeros: u8, _required_leading_zeros: u8, _block_hash: &[u8]) {
```

Comment:
Can you pass the block_hash as Sha256dHash instead of a [u8]? Ensures it can't accidentally get screwed and end up with a non-32-byte-slice at compile-time.

Reply:
Sure